poisson_lib.py

```
@@ -0,0 +1,731 @@
+        df: Input DataFrame
+        var_meta: Variable metadata dictionary
+        offset_col: Optional column name for exposure offset (e.g., person-years)
+        interaction_pairs: List of tuples for interactions [(var1, var2), ...]
+    
+    Returns:
+        tuple: (html_table, irr_results, airr_results, interaction_results)
+    """
+    from logic import (clean_numeric_value, _robust_sort_key, get_label, 
+                       fmt_p_with_styling)
+    
+    logger.info(f"Starting Poisson analysis for outcome: {outcome_name}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)